The job **`mingw-check-tidy`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8705349260/job/23875567568) [(plain)](https://github.com/rust-lang/rust/commit/df3d981c991e3601356a807f970c38e91c33e4be/checks/23875567568/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Getting action download info
Download action repository 'msys2/setup-msys2@v2.22.0' (SHA:cc11e9188b693c2b100158c3322424c4cc1dadea)
Download action repository 'actions/checkout@v4' (SHA:b4ffde65f46336ab88eb53be808477a3936bae11)
Download action repository 'actions/upload-artifact@v4' (SHA:5d5d22a31266ced268874388b861e4b58bb5c2f3)
Complete job name: PR - mingw-check-tidy
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
---
COPY scripts/sccache.sh /scripts/
RUN sh /scripts/sccache.sh

COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt \
    && pip3 install virtualenv
COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
COPY host-x86_64/mingw-check/validate-error-codes.sh /scripts/

# NOTE: intentionally uses python2 for x.py so we can test it still works.
# NOTE: intentionally uses python2 for x.py so we can test it still works.
# validate-toolstate only runs in our CI, so it's ok for it to only support python3.
ENV SCRIPT TIDY_PRINT_DIFF=1 python2.7 ../x.py test \
           --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
# This file is autogenerated by pip-compile with Python 3.10
# by the following command:
#
#    pip-compile --allow-unsafe --generate-hashes reuse-requirements.in
---

#12 [5/8] COPY host-x86_64/mingw-check/reuse-requirements.txt /tmp/
#12 DONE 0.0s

#13 [6/8] RUN pip3 install --no-deps --no-cache-dir --require-hashes -r /tmp/reuse-requirements.txt     && pip3 install virtualenv
#13 0.471   Downloading binaryornot-0.4.4-py2.py3-none-any.whl (9.0 kB)
#13 0.487 Collecting boolean-py==4.0
#13 0.494   Downloading boolean.py-4.0-py3-none-any.whl (25 kB)
#13 0.511 Collecting chardet==5.1.0
---
#13 3.707 Building wheels for collected packages: reuse
#13 3.708   Building wheel for reuse (pyproject.toml): started
#13 4.031   Building wheel for reuse (pyproject.toml): finished with status 'done'
#13 4.032   Created wheel for reuse: filename=reuse-1.1.0-cp310-cp310-manylinux_2_35_x86_64.whl size=181117 sha256=f5f58750481f69515c2c0d1d503daf565e2565c370d07fc6aeb95fe3498b4269
#13 4.032   Stored in directory: /tmp/pip-ephem-wheel-cache-da9c2_59/wheels/c2/3c/b9/1120c2ab4bd82694f7e6f0537dc5b9a085c13e2c69a8d0c76d
#13 4.035 Installing collected packages: boolean-py, binaryornot, setuptools, reuse, python-debian, markupsafe, license-expression, jinja2, chardet
#13 4.058   Attempting uninstall: setuptools
#13 4.059     Found existing installation: setuptools 59.6.0
#13 4.060     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.060     Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
#13 4.061     Can't uninstall 'setuptools'. No files were found to uninstall.
#13 4.734 Successfully installed binaryornot-0.4.4 boolean-py-4.0 chardet-5.1.0 jinja2-3.1.2 license-expression-30.0.0 markupsafe-2.1.1 python-debian-0.1.49 reuse-1.1.0 setuptools-66.0.0
#13 4.734 WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
#13 5.251 Collecting virtualenv
#13 5.307   Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
#13 5.598      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 13.1 MB/s eta 0:00:00
#13 5.639 Collecting distlib<1,>=0.3.7
#13 5.646   Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
#13 5.698      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 KB 9.1 MB/s eta 0:00:00
#13 5.733 Collecting platformdirs<5,>=3.9.1
#13 5.741   Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
#13 5.776 Collecting filelock<4,>=3.12.2
#13 5.783   Downloading filelock-3.13.4-py3-none-any.whl (11 kB)
#13 5.870 Installing collected packages: distlib, platformdirs, filelock, virtualenv
#13 6.057 Successfully installed distlib-0.3.8 filelock-3.13.4 platformdirs-4.2.0 virtualenv-20.25.1
#13 DONE 6.1s

#14 [7/8] COPY host-x86_64/mingw-check/validate-toolstate.sh /scripts/
#14 DONE 0.0s
---
DirectMap4k:      194496 kB
DirectMap2M:     9242624 kB
DirectMap1G:     9437184 kB
##[endgroup]
Executing TIDY_PRINT_DIFF=1 python2.7 ../x.py test            --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
+ TIDY_PRINT_DIFF=1 python2.7 ../x.py test --stage 0 src/tools/tidy tidyselftest --extra-checks=py:lint
    Finished `dev` profile [unoptimized] target(s) in 0.03s
##[endgroup]
downloading https://ci-artifacts.rust-lang.org/rustc-builds-alt/ad18fe08de03fbb459c05475bddee22707b4f0ec/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz
extracting /checkout/obj/build/cache/llvm-ad18fe08de03fbb459c05475bddee22707b4f0ec-true/rust-dev-nightly-x86_64-unknown-linux-gnu.tar.xz to /checkout/obj/build/x86_64-unknown-linux-gnu/ci-llvm
---
   Compiling tidy v0.1.0 (/checkout/src/tools/tidy)
    Finished `release` profile [optimized] target(s) in 25.21s
##[endgroup]
fmt check
##[error]Diff in /checkout/library/alloc/src/collections/binary_heap/mod.rs at line 1215:
     /// io::sink().write(heap.as_slice()).unwrap();
     #[must_use]
     #[must_use]
-    #[stable(feature = "binary_heap_as_slice",  since = "CURRENT_RUSTC_VERSION")]
+    #[stable(feature = "binary_heap_as_slice", since = "CURRENT_RUSTC_VERSION")]
     pub fn as_slice(&self) -> &[T] {
         self.data.as_slice()
     }
Running `"/checkout/obj/build/x86_64-unknown-linux-gnu/rustfmt/bin/rustfmt" "--config-path" "/checkout" "--edition" "2021" "--unstable-features" "--skip-children" "--check" "/checkout/library/alloc/src/ffi/c_str/tests.rs" "/checkout/library/alloc/src/ffi/mod.rs" "/checkout/library/alloc/src/string.rs" "/checkout/library/alloc/src/macros.rs" "/checkout/library/alloc/src/borrow.rs" "/checkout/library/alloc/src/lib.rs" "/checkout/library/alloc/src/slice/tests.rs" "/checkout/library/alloc/src/alloc.rs" "/checkout/library/alloc/src/testing/rng.rs" "/checkout/library/alloc/src/testing/mod.rs" "/checkout/library/alloc/src/testing/crash_test.rs" "/checkout/library/alloc/src/testing/ord_chaos.rs" "/checkout/library/alloc/src/task.rs" "/checkout/library/alloc/src/collections/binary_heap/tests.rs" "/checkout/library/alloc/src/collections/binary_heap/mod.rs" "/checkout/library/alloc/src/collections/linked_list/tests.rs" "/checkout/library/alloc/src/collections/mod.rs" "/checkout/library/alloc/src/vec/into_iter.rs" "/checkout/library/alloc/src/vec/set_len_on_drop.rs" "/checkout/library/alloc/src/vec/is_zero.rs" "/checkout/library/alloc/src/vec/in_place_drop.rs" "/checkout/library/alloc/src/vec/partial_eq.rs" "/checkout/library/alloc/src/vec/extract_if.rs" "/checkout/library/alloc/src/vec/mod.rs" "/checkout/library/alloc/src/vec/splice.rs" "/checkout/library/alloc/src/vec/cow.rs" "/checkout/library/alloc/src/vec/spec_from_iter_nested.rs" "/checkout/library/alloc/src/vec/drain.rs" "/checkout/library/alloc/src/vec/in_place_collect.rs" "/checkout/library/alloc/src/vec/spec_extend.rs" "/checkout/library/alloc/src/vec/spec_from_elem.rs" "/checkout/library/alloc/src/vec/spec_from_iter.rs" "/checkout/library/alloc/src/rc/tests.rs" "/checkout/library/alloc/src/sync/tests.rs" "/checkout/library/alloc/src/raw_vec.rs" "/checkout/library/alloc/src/sync.rs" "/checkout/library/alloc/src/fmt.rs" "/checkout/library/alloc/src/raw_vec/tests.rs" "/checkout/library/alloc/src/boxed.rs" "/checkout/library/alloc/src/lib.miri.rs" "/checkout/library/alloc/src/boxed/thin.rs" "/checkout/library/alloc/src/collections/vec_deque/into_iter.rs" "/checkout/library/alloc/src/collections/vec_deque/tests.rs" "/checkout/library/alloc/src/collections/vec_deque/macros.rs" "/checkout/library/alloc/src/collections/vec_deque/mod.rs" "/checkout/library/alloc/src/collections/vec_deque/drain.rs" "/checkout/library/alloc/src/collections/vec_deque/iter.rs" "/checkout/library/alloc/src/collections/vec_deque/spec_extend.rs" "/checkout/library/alloc/src/collections/vec_deque/iter_mut.rs" "/checkout/library/alloc/src/collections/vec_deque/spec_from_iter.rs" "/checkout/library/alloc/benches/slice.rs" "/checkout/library/alloc/benches/vec.rs" "/checkout/library/alloc/benches/string.rs" "/checkout/library/alloc/benches/lib.rs" "/checkout/library/alloc/benches/vec_deque_append.rs" "/checkout/library/alloc/benches/binary_heap.rs" "/checkout/library/alloc/benches/vec_deque.rs" "/checkout/library/alloc/benches/btree/map.rs" "/checkout/library/alloc/benches/btree/mod.rs" "/checkout/library/alloc/benches/btree/set.rs" "/checkout/library/alloc/benches/str.rs" "/checkout/library/alloc/benches/linked_list.rs" "/checkout/library/alloc/src/collections/btree/map.rs" "/checkout/library/alloc/src/alloc/tests.rs"` failed.
If you're running `tidy`, try again with `--bless`. Or, if you just want to format code, run `./x.py fmt` instead.
  local time: Tue Apr 16 12:16:09 UTC 2024
  network time: Tue, 16 Apr 2024 12:16:09 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/library/alloc/src/collections/binary_heap/mod.rs
+++ b/library/alloc/src/collections/binary_heap/mod.rs
@@ -1207,7 +1207,6 @@ impl<T, A: Allocator> BinaryHeap<T, A> {
     /// Basic usage:
     ///
     /// ```
-    /// #![feature(binary_heap_as_slice)]
     /// use std::collections::BinaryHeap;
     /// use std::io::{self, Write};
     ///
@@ -1216,7 +1215,7 @@ impl<T, A: Allocator> BinaryHeap<T, A> {
     /// io::sink().write(heap.as_slice()).unwrap();
     /// ```
     #[must_use]
-    #[unstable(feature = "binary_heap_as_slice", issue = "83659")]
+    #[stable(feature = "binary_heap_as_slice", since = "CURRENT_RUSTC_VERSION")]
     pub fn as_slice(&self) -> &[T] {
         self.data.as_slice()
     }

--- a/library/alloc/tests/lib.rs
+++ b/library/alloc/tests/lib.rs
@@ -26,7 +26,6 @@
 #![feature(binary_heap_into_iter_sorted)]
 #![feature(binary_heap_drain_sorted)]
 #![feature(slice_ptr_get)]
-#![feature(binary_heap_as_slice)]
 #![feature(inplace_iteration)]
 #![feature(iter_advance_by)]
 #![feature(iter_next_chunk)]

